### ISSUE TITLE
Import rewrite issue with `TypedDict`

### ISSUE THREAD

**Issue by cdce8p** [asottile/pyupgrade#683]
In Python `3.11` `TypedDict` can inherit from `Generic` which hasn't been possible so far. To use it in prior versions, `TypedDict` can be imported from `typing_extensions`. I believe it would make sense to postpone import rewriting for it until `3.11`.

Example from the Python [docs](https://docs.python.org/3.11/library/typing.html#typing.TypedDict)
```py
from typing import Generic, TypeVar
from typing_extensions import TypedDict

T = TypeVar("T")

class Group(TypedDict, Generic[T]):
    key: T
    group: list[T]
```

After `pyupgrade --py38-plus`
```py
from typing import Generic, TypeVar
from typing import TypedDict
...
```

--
Besides `TypedDict`, `NamedTuple` will also be able to inherit from `Generic` in `3.11` or can be imported from `typing_extensions`. This isn't an issue yet though, since pyupgrade doesn't rewrite it.

```py
from typing import Generic, TypeVar
from typing_extensions import NamedTuple

T = TypeVar("T")

class Group(NamedTuple, Generic[T]):
    key: T
    group: list[T]
```

**Comment by asottile** [comment]
makes sense -- want to send a PR to the generator in reorder-python-imports? (where this data is sourced from)

**Comment by asottile** [comment]
via #684
